client.go

```diff
@@ -618,17 +618,10 @@ func (c *Client) GetReadyMessagesCount(queueName string) (msgCount int, err erro
 	return
 }
 
-// ConsumeReadyMessages fetches given number of messages from queue
-func (c *Client) ConsumeReadyMessages(queueName string, numberOfMessages int) (msgs []amqp.Delivery, err error) {
-	ch, err := c.getChannel()
-	if err != nil {
-		log.Error().Err(err).Msg("c.getChannel()")
-		return nil, err
-	}
-	defer ch.Close()
-
-	for i := 0; i < numberOfMessages; i++ {
-		d, ok, err := ch.Get(queueName, false)
+// GetReadyMessages fetches given number of messages from queue
+func (c *Client) GetReadyMessages(queueName string, n int) (msgs []amqp.Delivery, err error) {
+	for i := 0; i < n; i++ {
+		d, ok, err := c.channel.Get(queueName, false)
```

Comment:
that is forbidden, you do not synchronize access and `channel` can not be used concurrently

Reply:
reverted code